code/modules/research/xenobiology/crossbreeding/warping_runes.dm

```diff
@@ -0,0 +1,309 @@
+
+/obj/effect/slimerune/grey/examine(mob/user)
+	..()
+	var/obj/item/slimecross/warping/grey/ex = extract
+	to_chat(user, "<span class='notice'>It has absorbed [ex.num_absorbed] extract[(ex.num_absorbed != 1) ? "s":""].</span>")
+	to_chat(user, "<span class='notice'>At 8 extracts, it can be manually fed an extract to produce a slime of that color.</span>")
+
+/obj/effect/slimerune/grey/process()
+	var/obj/item/slimecross/warping/grey/ex = extract
+	if(ex.num_absorbed >= 8)
+		return
+	if(locate(/obj/item/slime_extract) in get_turf(loc))
+		var/eaten = 0
+		for(var/obj/item/slime_extract/x in get_turf(src))
+			if(ex.num_absorbed >= 8)
+				break
+			qdel(x)
+			ex.num_absorbed++
+			eaten += 1
+		visible_message("<span class='notice'>[src] glows, and the extract[eaten != 1 ? "s":""] on it fade[eaten != 1 ? "":"s"] away.</span>")
+		if(ex.num_absorbed >= 8)
+			visible_message("<span class='notice'>[src] has finished its bluespace calibration, and can now produce a slime.</span>")
+
+/obj/effect/slimerune/grey/attackby(obj/item/slime_extract/O, mob/user)
+	var/obj/item/slimecross/warping/grey/ex = extract
+	if(!istype(O))
+		return
+	if(ex.num_absorbed < 8)
+		to_chat(user, "<span class='warning'>[src] isn't calibrated with enough slime extracts to produce a slime.</span>")
+		return
+	ex.num_absorbed = 0
+	user.visible_message("<span class='warning'>[user] places a slime extract onto [src], and a slime grows from it!</span>", "<span class='notice'>You place [O] onto [src], and a slime grows from it!</span>")
+	var/mob/living/simple_animal/slime/S = new(loc)
+	S.set_colour(O.slimecolor)
+	qdel(O)
+	return
+
+/obj/effect/slimerune/orange
+	name = "orange rune"
+	desc = "It's hot to the touch..."
+	var/obj/structure/bonfire/prelit/slime/fire
+
+/obj/effect/slimerune/orange/on_place()
+	fire = new(loc)
+
+/obj/effect/slimerune/orange/on_remove()
+	qdel(fire)
+
+/obj/structure/bonfire/prelit/slime
+	name = "magical flame"
+	desc = "It floats above the rune with constrained purpose."
+
+/obj/structure/bonfire/prelit/slime/CheckOxygen()
+	return TRUE
+
+/obj/structure/bonfire/prelit/slime/extinguish()
+	return
+
+/obj/effect/slimerune/purple
+	name = "purple rune"
+	desc = "It longs for cloth to weave and plastics to catalyze."
+
+/obj/effect/slimerune/purple/process()
+	for(var/obj/item/stack/sheet/cloth/C in get_turf(loc))
+		if(C.use(20))
+			var/obj/item/stack/medical/bruise_pack/B = new(loc)
+			visible_message("<span class='notice'>[src] weaves [C] into [B].")
+			break
+
+	for(var/obj/item/stack/sheet/plastic/P in get_turf(loc))
+		if(P.use(20))
+			var/obj/item/stack/medical/ointment/O = new(loc)
+			visible_message("<span class='notice'>[src] melts down [P], creating [O].")
+			break
+
+/obj/effect/slimerune/blue
+	name = "blue rune"
+	desc = "It constantly mists water from its intricate design."
+
+/obj/effect/slimerune/blue/ComponentInitialize()
+	AddComponent(/datum/component/slippery, 80)
```

Comment:
`. = ..()` at the top

Also this proc is a bit outdated and you can have these in regular init if you wish